src/debug-generator/debug-generator.c

```diff
@@ -66,6 +66,17 @@ static const struct BreakpointInfo breakpoint_info_table[_BREAKPOINT_TYPE_MAX] =
         { BREAKPOINT_PRE_SWITCH_ROOT,   "pre-switch-root", "breakpoint-pre-switch-root.service", BREAKPOINT_IN_INITRD | BREAKPOINT_DEFAULT },
 };
 
+static bool breakpoint_applies(const BreakpointInfo *info, int log_level) {
+        if (in_initrd() && !FLAGS_SET(info->validity, BREAKPOINT_IN_INITRD))
+                log_full(log_level, "Breakpoint '%s' not valid in the initrd, ignoring.", info->name);
+        else if (!in_initrd() && !FLAGS_SET(info->validity, BREAKPOINT_ON_HOST))
+                log_full(log_level, "Breakpoint '%s' not valid on the host, ignoring.", info->name);
+        else
+                return true;
+
+        return false;
```

Comment:
This is not reachable?

Reply:
It's reachable, there is no default breakpoint on the host.

Reply:
Ignore me I'm stupid